### PR TITLE
Migrate FA processor to SDK

### PR DESCRIPTION
**Pull request by rtso**
## Description

- Upgrade SDK dependency to latest version
- Migrate FungibleAssetProcessor to
  - Add FungibleAssetExtractor and FungibleAssetStorer steps
  - Update config to support FungibleAssetProcessor
  - Add support for deprecated tables flags

## Testing
Run fa processor

### CODE REVIEW

**Comment by rtso** [comment]
* **#543** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/543?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rtso and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: rust/sdk-processor/src/config/indexer_processor_config.rs [hunk 1]
```diff
@@ -2,18 +2,25 @@
...
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct IndexerProcessorConfig {
     pub processor_config: ProcessorConfig,
     pub transaction_stream_config: TransactionStreamConfig,
     pub db_config: DbConfig,
+
+    // String vector for deprecated tables to skip db writes
+    #[serde(default)]
+    pub deprecated_tables: HashSet<String>,
```
Comment (grao1991) [inline rust/sdk-processor/src/config/indexer_processor_config.rs:23]: This probably should be put inside each individual processor config, not at the top level.

File: rust/sdk-processor/src/processors/fungible_asset_processor.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    common_steps::TransactionStreamStep,
+    traits::IntoRunnableStep,
+};
+use processor::worker::TableFlags;
+use serde::{Deserialize, Serialize};
+use tracing::{debug, info};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct FungibleAssetProcessorConfig {
```
Comment (dermanyang) [inline rust/sdk-processor/src/processors/fungible_asset_processor.rs:33]: Can this be abstracted out in someway? looks the same as the `EventsProcessorConfig`

File: rust/sdk-processor/src/processors/fungible_asset_processor.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            DbConfig::PostgresConfig(ref postgres_config) => {
+                run_migrations(
+                    postgres_config.connection_string.clone(),
+                    self.db_pool.clone(),
+                )
+                .await;
+            },
+        }
+
+        // (Optional) Merge the starting version from config and the latest processed version from the DB
```
Comment (dermanyang) [inline rust/sdk-processor/src/processors/fungible_asset_processor.rs:95]: What does optional refer to here?

File: rust/sdk-processor/src/processors/fungible_asset_processor.rs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        let starting_version = get_starting_version(&self.config, self.db_pool.clone()).await?;
+
+        // (Optional) Check and update the ledger chain id to ensure we're indexing the correct chain
```
Comment (dermanyang) [inline rust/sdk-processor/src/processors/fungible_asset_processor.rs:98]: What does optional refer to here/why is it not necessary?

**Comment by dermanyang** [comment]
If we're thinking about adding documentation as we migrate these processors over to the SDK, it would be nice if we standardize where we write that documentation. maybe just in the struct definition in `fungible_asset_processor.rs` . Which steps are needed/what's unique to this processor oslt. What do we think?

**Review by dermanyang — APPROVED** [review]

**Comment by rtso** [comment]
@dermanyang good call, I'll add some documentation in the next PR. I think the documentation should be inside each step. FungibleAssetExtractor includes most of the processing logic, and we should document each loop and what is being extracted.

File: rust/sdk-processor/src/processors/fungible_asset_processor.rs (hunk 2 above)
Comment (rtso) [reply rust/sdk-processor/src/processors/fungible_asset_processor.rs]: Abstracted out into DefaultProcessorConfig

File: rust/sdk-processor/src/config/indexer_processor_config.rs (hunk 1 above)
Comment (rtso) [reply rust/sdk-processor/src/config/indexer_processor_config.rs]: Moved it to individual processor config

**Comment by rtso** [comment]
### Merge activity

* **Oct 8, 5:07 PM EDT**: @rtso merged this pull request with [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processors/543).

**Pull request merged**